### PR TITLE
PICARD-1228: Enable remove button for unclustered files only when applicable

### PR DESCRIPTION
**Pull request by vishichoudhary**
# Summary

* This is a…
    * [x] Bug fix
    * [ ] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem
Remove button is still enabled if there isn't anything to remove in the unclustered file. Similarly, there is expand all and collapse all option in the context menu if there isn't anything to expand and collapse.

* JIRA ticket (_optional_): [PICARD-1228](https://tickets.metabrainz.org/browse/PICARD-1228)

# Solution

# Action

### CODE REVIEW

**Comment by vishichoudhary** [comment]
@zas . Done,

**Comment by vishichoudhary** [comment]
@zas Something wrong with Travis-ci. Been stuck for quite a long time

**Comment by zas** [comment]
> @zas Something wrong with Travis-ci. Been stuck for quite a long time

@vishichoudhary : yes, not sure what is going on, can you just rebase and push ?

File: picard/ui/mainwindow.py [hunk 1]
```diff
@@ -818,7 +818,12 @@ def save(self):
 
     def remove(self):
         """Tell the tagger to remove the selected objects."""
-        self.panel.remove(self.selected_objects)
+        if isinstance(self.selected_objects[0], ClusterList):
```
Comment (sambhav) [inline picard/ui/mainwindow.py]: This is the completely wroing place to add remove logic. All the remove logic exists in tagger,py#607.

Move this to that. Probably what is causing the spotty bugs.

File: picard/ui/itemviews.py [hunk 2]
```diff
@@ -399,8 +400,7 @@ def _add_other_versions():
                     else:
                         action_menu = plugin_menus[key] = action_menu.addMenu(key[-1])
                 action_menu.addAction(action)
-
-        if isinstance(obj, Cluster) or isinstance(obj, ClusterList) or isinstance(obj, Album):
+        if (isinstance(obj, Cluster) and len(obj.files) > 0) or (isinstance(obj, ClusterList) and len(obj) > 0) or isinstance(obj, Album):
```
Comment (sambhav) [inline picard/ui/itemviews.py]: You might wanna break up the if logic into multiple lines. This is hard to read.

**Comment by sambhav** [comment]
I fail to see how the context menu changes are related to the ticket directly. You want to file another ticket and move these changes to another PR.

**Comment by vishichoudhary** [comment]
@sambhav i have split the pr and will fix the context menu issues in the next pr.

**Comment by sambhav** [comment]
@vishichoudhary I rebased your commit and reworded this. This is the last time I am going to point out the contribution guidelines for you, I have already done so multiple times. If you cannot adhere to them, I wont be accepting any PRs from you.

> Make your changes and commit them with a good description. Your commit subject should be written in **imperative voice** and **sentence case**.

The important parts are even bolded! Make sure that all your commit messages follow the guidelines, its not that tough.

**Pull request merged**